### PR TITLE
docs: add documentation for the courier.message_retries key

### PR DESCRIPTION
**Pull request by jonas-jonas**
## Related Issue or Design Document

https://github.com/ory/kratos/pull/2638

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [x] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: docs/kratos/concepts/email-sms.md [hunk 1]
```diff
@@ -394,3 +394,18 @@ courier:
 {{- end -}}
 
 ```
+
+## Handling errors
```
Comment (aeneasr) [inline docs/kratos/concepts/email-sms.md:398]: @tomekpapiernik fyi this is currently open-source only as we limit the retries in cloud to exactly 5 and don't want to make it configurable.

**Review by aeneasr — COMMENTED** [review]
From my side 👍

**Review by aeneasr — APPROVED** [review]

**Comment by tomekpapiernik** [comment]
@jonas-jonas I'm currently preparing a PR that includes major changes to this file here. Would you mind closing this PR? I'll include the paragraph you added in my pool of changes. This will save me some time as I won't have to deal with conflict solving (at least I'll have one file less to worry about) 😅.

**Comment by jonas-jonas** [comment]
@tomekpapiernik sure! No problem. :)

**Comment by tomekpapiernik** [comment]
Thank you!

**Pull request closed without merging**